pelican/server.py

```diff
@@ -23,16 +24,17 @@ def do_GET(self):
         for suffix in SUFFIXES:
             if not hasattr(self,'original_path'):
                 self.original_path = self.path
-            self.path = self.original_path + suffix
-            path = self.translate_path(self.path)
-            if os.path.exists(path):
-                srvmod.SimpleHTTPRequestHandler.do_GET(self)
-                logging.info("Found: %s" % self.path)
-                found = True
-                break
-            logging.info("Tried to find file %s, but it doesn't exist. ", self.path)
+            for strip_pattern in RSTRIP_PATTERNS:
+                self.path = self.original_path.rstrip(strip_pattern) + suffix
+                path = self.translate_path(self.path)
+                if os.path.exists(path):
+                    srvmod.SimpleHTTPRequestHandler.do_GET(self)
+                    logging.info("Found: %s" % self.path)
+                    found = True
+                    break
```

Comment:
Thank you for the review! This code has fallen behind so I updated it with the recent pelican. Replacing the for...else logic with an explicit "found" flag now since we have a double loop. The strip pattern loop "prefers" no stripping to ensure we find the original path first.